MH-23 Add home page

dsummersl opened the pull request:
This adds some additional docker-centric tooling to ensure that the project will work in staging/production environments:

 * npm packages are added to the STATICFILES_DIRS and installed inside
   the Dockerfile-based container, so that when 'collectstatic' is run, it
   finds the files in the container (not the local dev environment mounted
   into the container). When new npm packages are installed, the
   developer will need to run './hsctl rebuild' to ensure that the
   packages are added to the container they are working on (this was/is already true for pip dependencies).
 * I broke out all the 'new for myhpom' pip packages into a standard
   requirements.txt file.
 * Created a base.html template.
 * I moved all the original endpoints to /hydroshare/ (site still works, from what I can tell), while the home page is the new MyHPOM home page.
 * Adds /styleguide using http://astrum.nodividestudio.com

Additional notes:
 * make sure to rerun `./hysctl rebuild --db` as some mezzanine pages (my-documents, etc) slugs are updated.
 * Since the styleguide is just static content, you must manually run 'collectstatic' whenever you add new components. The `astrum` command line tool has been added to the django docker image so you can add new components within it with a command like `docker-compose run --rm hydroshare astrum new buttons/foobar` command.

dsummersl committed:
Adds new home page and moves hydroshare into /hydroshare URLs (MH-23)

Existing hydroshare pages will continue to be accessible for during the
development phase, but before our production release we will want to
disable it.

dsummersl committed:
Fixes /userInfo call made from the home page (MH-23)

dsummersl committed:
Adds bootstrap and initial home page styling (MH-23).

This adds some additional docker-centric tooling to ensure that the
project will work in staging/production environments:
 * npm packages are added to the STATICFILES_DIRS and installed inside
   the Dockerfile-based container, so that when 'collectstatic' is run, it
   finds the files in the container (not the local dev environment mounted
   into the container). When new npm packages are installed, the
   developer will need to run './hsctl rebuild' to ensure that the
   packages are added to the container they are working on.
 * I broke out all the 'new for myhpom' pip packages into a standard
   requirements.txt file.
 * Created a base.html template.

dsummersl committed:
Adds login button to home page (MH-23)

dsummersl committed:
Removes non-icon fb/ln links (MH-23)

dsummersl committed:
Changes all hardcoded hydroshare urls to start in /hydroshare (MH-23)

If these had all been set up with reverse() this wouldn't have had to
happen. Since we don't plan on supporting this in the long term, I don't
think its worth the significant extra work to map these to actual url
names.

dsummersl committed:
Adds myhpom stylesheet (MH-23)

dsummersl committed:
Adds compilescss for staging/production environments (MH-23)

Also make the nginx script bash v3 compatible by removing the auto-lowercase
syntax (OSX doesn't have that version):

http://wiki.bash-hackers.org/syntax/pe#case_modificationarrays

dsummersl committed:
Changes many /hsapi paths to /hydroshare/hsapi (MH-23)

dsummersl committed:
Adds astrum example styleguide (MH-23)

dsummersl committed:
Removes unrelated content from top of readme (MH-23)

dsummersl committed:
Fixes create new version path change (MH-23)

The parsed response didn't account for the new level of nesting.

dsummersl committed:
Fixes values that are parsed from URLs (MH-23)

Since all the /hydroshare URLs are one level further into the URL path,
several tests broke b/c the IDs that the tests were looking for have
been moved a little further down. Ideally this code would have fetched
the ID from another reliable backend source, rather than parsing it out
of the URL.

dsummersl committed:
Fixes test case paths (MH-23)

dsummersl committed:
Fixes test case paths (MH-23)

nmashton commented on hydroshare/settings.py:
```diff
@@ -313,8 +317,15 @@
     "hs_rest_api",
     "hs_dictionary",
     "security",
+    "sass_processor",
```
I like this, but one thing that I'm not seeing is: sourcemaps. Did you get that to work? Did you have to do anything special to do so?

dsummersl replied:
you know...I see source maps genereted in the static dirs, but I didn't bother to test them. I'll take a poke at that.

dsummersl replied:
they are generated - but the paths look wrong. It looks like *maybe* this will be fixed in the next release: https://github.com/jrief/django-sass-processor/pull/96

nmashton commented on myhpom/templates/myhpom/base.html:
```diff
@@ -0,0 +1,93 @@
+                {% block main_content %}{% endblock main_content %}
+            </div>
+        </main>
+    </body>
```
This should probably be lower! :)

dsummersl replied:
doh!

nmashton commented on package.json:
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "myhpom",
+  "version": "0.0.1",
+  "description": "",
```
Could we flesh this out with some text copied from the repo?

nmashton reviewed (changes requested):
There's one important slip in the main template HTML that needs fixing. Otherwise, I like it all.

It looks like the overwhelming majority of changes are prefixing URLs, which I don't object to. I trust that eventually you'll nail down all these cases and tests will all pass. :)

One of my comments was worked out in practice: fix that `urlpatterns` so that MyHPOM's patterns are recognized. You did that.

Another of my "comments" would be: we should use the old stretchy-`main` trick to ensure that the footer is always glued to the bottom of the page. But that change is made in my branch, so it'll happen soon anyway.

dsummersl committed:
Puts <footer> in body (MH-23)

Woops!

dsummersl committed:
Adds description to npm package name (MH-23)

nmashton reviewed (approved):
Make it so 🍵

dsummersl committed:
Fixes old hydroshare paths (MH-23)

dsummersl committed:
Fixes mezzanine page resolution (MH-23)

Some tests were breaking because the Pages were resolving as
/hydroshare/hydroshare/foo.
